File: app/src/main/java/com/example/epamtraining/activities/TrainingsActivity.java
package com.example.epamtraining.activities;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageButton;

import com.example.epamtraining.Callbacks.SwipeToDeleteExerciseCallback;
import com.example.epamtraining.R;
import com.example.epamtraining.adapters.TrainingsAdapter;
import com.example.epamtraining.models.Exercises;

public class TrainingsActivity extends AppCompatActivity {

    private ImageButton addExerciseButton;
    private EditText exerciseNameText;
    private EditText exerciseSetsText;

    private RecyclerView recyclerView;
    private TrainingsAdapter trainingsAdapter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_trainings);

        addExerciseButton = findViewById(R.id.add_exercise_button);
        exerciseNameText = findViewById(R.id.exercise_name_text);
        exerciseSetsText = findViewById(R.id.exercise_sets_text);

        recyclerView = findViewById(R.id.exercises);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        trainingsAdapter = new TrainingsAdapter();
        recyclerView.setAdapter(trainingsAdapter);
        ItemTouchHelper itemTouchHelper = new
                ItemTouchHelper(new SwipeToDeleteExerciseCallback(trainingsAdapter));
        itemTouchHelper.attachToRecyclerView(recyclerView);

        addExerciseButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = exerciseNameText.getText().toString();
                int sets = Integer.parseInt(exerciseSetsText.getText().toString());
                trainingsAdapter.setItem(new Exercises(name,sets));
                exerciseNameText.setText(null);
                exerciseSetsText.setText(null);
            }
        });
    }
}

File: app/src/main/java/com/example/epamtraining/activities/TrainingsActivity.kt
package com.example.epamtraining.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.helper.ItemTouchHelper
import com.example.epamtraining.Callbacks.ExerciseTouchCallback
import com.example.epamtraining.R
import com.example.epamtraining.adapters.TrainingsAdapter
import com.example.epamtraining.models.Exercises
import kotlinx.android.synthetic.main.fragment_trainings.*
import java.util.*

class TrainingsActivity : AppCompatActivity() {

    lateinit private var trainingsAdapter: TrainingsAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_trainings)

        exercisesRecyclerView?.layoutManager = LinearLayoutManager(this)
        trainingsAdapter = TrainingsAdapter()
        exercisesRecyclerView?.adapter = trainingsAdapter
        val itemTouchHelper = ItemTouchHelper(ExerciseTouchCallback(trainingsAdapter))
        itemTouchHelper.attachToRecyclerView(exercisesRecyclerView)
        exercisesRecyclerView?.itemAnimator = object : DefaultItemAnimator() {
            override fun animateMove(holder: RecyclerView.ViewHolder, fromX: Int, fromY: Int, toX: Int, toY: Int): Boolean {
                return super.animateMove(holder, fromX, fromY, toX, toY)
            }
        }
        exercisesRecyclerView?.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))

        addExerciseButton?.setOnClickListener {
            val name = exerciseNameText?.text.toString()
            val sets = Integer.parseInt(exerciseSetsText?.text.toString())
            trainingsAdapter.setItem(Exercises(UUID.randomUUID(), name, sets, 1.0))
            exerciseNameText?.setText(null)
            exerciseSetsText?.setText(null)
        }
    }
}
